src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -560,11 +562,18 @@ export class NotebookService extends Disposable implements INotebookService {
 		if (notebookUri.scheme === Schemas.untitled) {
 			return true;
 		}
+		const trustedBooksConfigKey = 'notebook.trustedBooks';
 
 		let cacheInfo = this.trustedNotebooksMemento.trustedNotebooksCache[notebookUri.toString()];
 		if (!cacheInfo) {
-			// This notebook was never trusted
-			return false;
+			// Check if the notebook belongs to a book that's trusted.
+			let trustedBookDirectories: string[] = this._configurationService?.getValue(trustedBooksConfigKey) ?? [];
+			if (trustedBookDirectories.find(b => notebookUri.fsPath.indexOf(b) > -1)) {
```

Comment:
Just because a Notebook is in a directory doesn't necessarily mean it's a part of that book, right? At least we need to make a decision about whether we assume that's always going to be the case - it doesn't seem unreasonable to make this assumption but we need to make sure we're all in agreement this is the direction we want to go.

Reply:
From the book tree view, that's the presumption: https://github.com/microsoft/azuredatastudio/blob/9aa4a03ecf4328e53aef78c62a5d64feaae6e0a9/extensions/notebook/src/book/bookTrustManager.ts#L29